docs/Conditions.md

```diff
@@ -191,4 +211,48 @@ Usage can then look as following:
 #else
 // This renders for desktop platforms
 #endif
-```
\ No newline at end of file
+```
+
+## Conditional Parameters
+
+[Parameter symbols in template](Reference-for-template.json.md#parameter-symbol) can be specified together with optional conditions:
+* [`Is Enabled Condition`](Reference-for-template.json.md#isEnabledCondition) - overwritting presence of input parameter. If condition is specified and evaluates to false, passed parameter value (if any) is ignored and processing works as if the symbol would not exist. This includes application of [default values](Reference-for-template.json.md#default), [verification of mandatory parameters](Reference-for-template.json.md#isRequired), [conditional processing of sources](Conditional-processing-and-comment-syntax.md) and [replacements](Reference-for-template.json.md#replaces).
+* [`Is Required Condition`](Reference-for-template.json.md#isRequiredCondition) - dictates if parameter is mandatory or optional. After evaluation the behavior is identical as for the [`isRequired`](Reference-for-template.json.md#isRequired) config.
+
+### Evaluation
+
+**Input** - currently only other parameter symbols from the template configuration are supported within the parameter conditions. Any other variables are not replaced.
+
+**Evaluation order** - Dependencies between parameters are detected and evaluation is peformed in order that guarantees that all dependencies are evaluated prior their dependant (see [Topological Sorting](https://en.wikipedia.org/wiki/Topological_sorting) for details).
+
+ In presence of cyclic dependency the evaluation proceeds only if current input values of parameters do not lead to undeterministic result (and the cycle is indicated in warning log message). Otherwise an error is reported, indicating the cycle.
+
+ Example `template.json` with cyclic dependency:
+```json
+  "symbols": {
+    "A": {
+      "type": "parameter",
+      "datatype": "bool",
+      "isEnabledCondition": "B != false",
+    },
+    "B": {
+      "type": "parameter",
+      "datatype": "bool",
+      "isEnabledCondition": "A != true",
+    }
+}
+```
+
+Following input parameter values can (and will) be evaluated deterministically: `--A false --B true`
+
+Following input parameter values cannot be evaluated deterministically (and will lead to error): `--A true --B false`
+
+**Applying of host and default values** - All host and default values are applied before the conditions evaluation. After the evaluation defaults are reapplied to parameters that were evaluated as optional and that do not have host supplied values. After this an evaluation of presence of mandatory values takes place.
+
+### Performing evaluation externally
```

Comment:
@phenning Phill - once onboarding onto conditional parameters feature - this will be a way how to skip the template engine internal evaluation and supply the results from VS